src/main/java/org/sefglobal/scholarx/service/CommentService.java

```diff
@@ -0,0 +1,121 @@
+package org.sefglobal.scholarx.service;
+
+import org.sefglobal.scholarx.exception.BadRequestException;
+import org.sefglobal.scholarx.exception.ResourceNotFoundException;
+import org.sefglobal.scholarx.exception.UnauthorizedException;
+import org.sefglobal.scholarx.model.Comment;
+import org.sefglobal.scholarx.model.Mentee;
+import org.sefglobal.scholarx.model.Mentor;
+import org.sefglobal.scholarx.model.Profile;
+import org.sefglobal.scholarx.repository.CommentRepository;
+import org.sefglobal.scholarx.repository.MenteeRepository;
+import org.sefglobal.scholarx.repository.MentorRepository;
+import org.sefglobal.scholarx.repository.ProfileRepository;
+import org.sefglobal.scholarx.util.ProfileType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CommentService {
+    private final static Logger log = LoggerFactory.getLogger(CommentService.class);
+    private final CommentRepository commentRepository;
+    private final ProfileRepository profileRepository;
+    private final MenteeRepository menteeRepository;
+    private final MentorRepository mentorRepository;
+
+    public CommentService(ProfileRepository profileRepository,MenteeRepository menteeRepository,
+                          MentorRepository mentorRepository,
+                                CommentRepository commentRepository){
+        this.commentRepository = commentRepository;
+        this.profileRepository = profileRepository;
+        this.mentorRepository = mentorRepository;
+        this.menteeRepository = menteeRepository;
+    }
+
+    public List<Comment> getAllMenteeComments(long menteeId, long profileId)
+            throws ResourceNotFoundException, BadRequestException, UnauthorizedException {
+        Optional<Mentee> optionalMentee = menteeRepository.findById(menteeId);
+        Optional<Profile> optionalProfile = profileRepository.findById(profileId);
+        List<Mentor> optionalMentor = mentorRepository.findAllByProfileId(profileId);
```

Comment:
Here we don't have to get a mentor list. In line 50 we can directly use,
`optionalMentee.get().getAssignedMentor().getProfile.getId()==profileId`

Same in `addMenteeComment` method